miqcli/collections/automation_requests.py

```diff
@@ -14,16 +14,81 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 
+import click
+from collections import OrderedDict
+
+
+from miqcli.constants import FLOATINGIP_PAYLOAD, SUPPORTED_AUTOMATE_TYPES, \
+    OS_NETWORK_TYPE, OS_TYPE
+
+from miqcli.utils import log, get_input_data
+
 from miqcli.decorators import client_api
 
 
 class Collections(object):
     """Automation requests collections."""
 
+    @click.option('--type', type=str,
+                  help='type of automation request',
+                  required=True)
+    @click.option('--payload', type=str,
+                  help='payload data for an automation request')
+    @click.option('--payload_file', type=str,
+                  help='file name of the payload data for an '
+                       'automation request')
     @client_api
-    def create(self):
-        """Create."""
-        raise NotImplementedError
+    def create(self, type, payload, payload_file):
+        """
+        Create an automation request
+
+        :param type: type of automation request
+        :param payload: dictionary in string format
+        :param payload_file: file location of the payload
+        :return: id of the automation request
+        """
+        log.info("Attempt to create an automation request")
+
+        # Verify the request is a supported request
+        if type not in SUPPORTED_AUTOMATE_TYPES:
+            log.abort("Invalid request, automate only supports options: "
+                      "{0}".format(SUPPORTED_AUTOMATE_TYPES))
+
+        # get the data from user input
+        input_data = get_input_data(payload, payload_file)
+
+        if type == "gen_floating_ip":
```

Comment:
Is this check happening even if I provision instances in other providers such as RHV, AWS, etc?

Reply:
The automation requests call right now is very specific to the Automate scripts we create, I currently only added the single one for Steve's creating a floating ip for OpenStack.  I should probably make it clear this is for OpenStack, and I should probably also modify this if type is not set, to assume everything is setup correclty and just send the payload to the Automate request to support generic usage of it.